substrate/frame/indices/src/lib.rs

```diff
@@ -232,6 +232,57 @@ pub mod pallet {
 			Self::deposit_event(Event::IndexFrozen { index, who });
 			Ok(())
 		}
+
+		/// Reconsider the deposit reserved for an index.
+		/// 
+		/// The dispatch origin for this call must be _Signed_ and the signing account must have a
+		/// non-frozen account `index`.
+		/// 
+		/// The transaction fees is waived if the deposit is changed after reconsideration.
+		/// 
+		/// - `index`: the index whose deposit is to be reconsidered.
+		/// 
+		/// Emits `DepositReconsidered` if successful.
+		/// 
+		/// ## Complexity
+		/// - `O(1)`.
+		#[pallet::call_index(5)]
+		#[pallet::weight(T::WeightInfo::reconsider())]
+		pub fn reconsider(origin: OriginFor<T>, index: T::AccountIndex) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			Accounts::<T>::try_mutate(index, |maybe_value| -> DispatchResultWithPostInfo {
+				let (account, old_amount, perm) = maybe_value.take().ok_or(Error::<T>::NotAssigned)?;
+				ensure!(!perm, Error::<T>::Permanent);
+				ensure!(account == who, Error::<T>::NotOwner);
```

Comment:
We can probably remove this check and allow anyone to adjust anyone else's deposit, but something for later.